RELEASE.rst

```diff
@@ -31,39 +31,74 @@ that changes the major version number.
 Doing a Release
 ---------------
 
-1. Test releases on Python 2.7 and 3.4+ on Windows, Linux and OSX,
-   with and without the C extensions. It's generally enough to just run the
-   tests on 2.7, 3.4 and the latest 3.x version with and without the
-   extensions on a single platform, and then just test any version on the
-   other platforms as a sanity check. `python setup.py test` will build the
-   extensions and test. `python tools/clean.py` will remove the extensions,
+1. PyMongo is tested on Evergreen. Ensure the latest commit are passing CI
+   as expected: https://evergreen.mongodb.com/waterfall/mongo-python-driver.
+   To test locally, `python setup.py test` will build the C extensions and
+   test. `python tools/clean.py` will remove the extensions,
    and then `python setup.py --no_ext test` will run the tests without
    them. You can also run the doctests: `python setup.py doc -t`.
 
-2. Add release notes to doc/changelog.rst. Generally just summarize/clarify
+2. Check Jira to ensure all the tickets in this version have been completed.
+
+3. Add release notes to doc/changelog.rst. Generally just summarize/clarify
    the git log, but you might add some more long form notes for big changes.
 
-3. Search and replace the "devN" version number w/ the new version number (see
+4. Search and replace the "devN" version number w/ the new version number (see
    note above).
 
-4. Make sure version number is updated in setup.py and pymongo/__init__.py
+5. Make sure version number is updated in setup.py and pymongo/__init__.py
+
+6. Commit with a BUMP version_number message, eg `git commit -m 'BUMP 3.11.0'`.
+
+7. Tag w/ version_number, eg, `git tag -a '3.11.0' -m '3.11.0' <COMMIT>`.
+
+8. Push commit / tag, eg `git push && git push --tags`.
+
+9. Pushing a tag will trigger a release process in Evergreen which builds
+   wheels and eggs for manylinux, macOS, and Windows. Wait for these jobs to
+   complete and then download the "Release files" archive from each task. See:
+   https://evergreen.mongodb.com/waterfall/mongo-python-driver?bv_filter=release
```

Comment:
Should this be a Spruce link instead to ensure it doesn't go stale soon? I am not sure if the evergreen URLs are going away but Spruce certainly seems to be the future.

Reply:
Spruce doesn't have a waterfall page yet, it only supports viewing patch builds.